PR: https://github.com/odoo/odoo/pull/

From: 68e9d5504cb293643f2fb48fa4cc9278f6ad2ae3
From: Jeremy Kersten
Date: 2018-03-02 17:03:01

Structural Changes: 8
Total Changes: 8

[IMP] website: make website.redirect fields required

Before this commit, it was possible to create redirect with empty values.
Now, for new db, that will be required.

Fix usability according to first feedback
   - Redirect / menu should be visible without debug mode
     because user need it when they start their website and
     dont know at the beginning how to enable it (fgi)
   - fa icon in page manager was for blind people
   - missing widget on sequence field

================================= pseudo patch: =================================

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -891,9 +891,9 @@ class WebsiteRedirect(models.Model):
     _order = "sequence, id"
     _rec_name = 'url_from'
 
-    type = fields.Selection([('301', 'Moved permanently'), ('302', 'Moved temporarily')], string='Redirection Type')
-    url_from = fields.Char('Redirect From')
-    url_to = fields.Char('Redirect To')
+    type = fields.Selection([('301', 'Moved permanently (301)'), ('302', 'Moved temporarily (302)')], string='Redirection Type', required=True)
+    url_from = fields.Char('Redirect From', required=True)
+    url_to = fields.Char('Redirect To', required=True)
     website_id = fields.Many2one('website', 'Website')
     active = fields.Boolean(default=True)
-    sequence = fields.Integer(default=0)
+    sequence = fields.Integer()
